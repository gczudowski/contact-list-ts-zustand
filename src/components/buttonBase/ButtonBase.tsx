import React, { ReactElement } from 'react';

interface Props {
  onClick: () => void;
  label: string;
}

function ButtonBase({ onClick, label }: Props): ReactElement {
  return (
    <button style={styles.buttonBase} onClick={onClick} type="button">
      {label}
    </button>
  );
}

const styles = {
  buttonBase: {
    display: 'flex',
    width: '100%',
    minHeight: 36,
    fontSize: 18,
    backgroundColor: '#e74c3c',
    color: 'white',
    border: 'none',
    alignItems: 'center',
    justifyContent: 'center',
    cursor: 'pointer',
    boxShadow: '0px 1px 2px 0px rgb(0 0 0 / 15%)',
  },
};

export default ButtonBase;
