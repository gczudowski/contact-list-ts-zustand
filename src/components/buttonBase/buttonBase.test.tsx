import React from 'react';
import renderer from 'react-test-renderer';
import ButtonBase from './ButtonBase';

describe('ButtonBase', () => {
  const labelMock = 'labelMock';
  const onClickCallback = () => {};

  it('renders properly with initial props', () => {
    const component = renderer.create(<ButtonBase onClick={onClickCallback} label={labelMock} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it(`calls onClick callback prop`, () => {
    const onClickMock = jest.fn();
    const { root } = renderer.create(<ButtonBase onClick={onClickMock} label={labelMock} />);

    root.props.onClick();

    expect(onClickMock).toHaveBeenCalledWith();
  });
});
