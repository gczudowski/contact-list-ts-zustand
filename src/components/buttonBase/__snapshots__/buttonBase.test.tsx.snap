// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBase renders properly with initial props 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e74c3c",
      "border": "none",
      "boxShadow": "0px 1px 2px 0px rgb(0 0 0 / 15%)",
      "color": "white",
      "cursor": "pointer",
      "display": "flex",
      "fontSize": 18,
      "justifyContent": "center",
      "minHeight": 36,
      "width": "100%",
    }
  }
  type="button"
>
  labelMock
</button>
`;
